exo-add/features/steps/steps-given.ls

```diff
@@ -17,3 +18,14 @@ module.exports = ->
     @app-dir := path.join process.cwd!, 'tmp'
     fs.empty-dir-sync @app-dir
 
+
+  @Given /^I am in the directory of an application containing a "([^"]*)" service$/, (service-role, done) !->
+    @app-dir := path.join process.cwd!, 'tmp', 'app'
+    @create-empty-app('app', done)
+      .then ~>
```

Comment:
You call `done` here and on line 31. It should be called only once. 
Also, please don't add a new paradigm like promise APIs to our code base. 
We use continuation-style callbacks, and might move to generators in the future.
Promises just add complexity and boilerplate without a clear added benefit.